Add 4.4-dev7 article (#980)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-4-dev-7.md

[4.4 dev 3](/article/dev-snapshot-godot-4-4-dev-3/), [4.4 dev 4](/article/dev-snapshot-godot-4-4-dev-4/), [4.4 dev 5](/article/dev-snapshot-godot-4-4-dev-5/), and [4.4 dev 6](/article/dev-snapshot-godot-4-4-dev-6/) release notes for an overview of
some key features which were already in those snapshots, and are therefore still available for testing in dev 7.

Here are highlights of a few new features in dev 7 that you might find particularly exciting!

### Jolt Physics Module

Ever since its inception in late 2022, [godot-jolt](https://github.com/godot-jolt/godot-jolt) has become the de-facto 3D physics engine for many of our developers. Much of the history behind why this was the case is documented in [this proposal](https://github.com/godotengine/godot-proposals/issues/7308) by [Adam Scott](https://github.com/adamscott), but the main takeaway was a strong push for this tool to be recognized as an official addon. In doing so, users would be able to find this amazing resource in a way that was promoted by the engine itself; an exciting prospect!

> The Godot Jolt maintainers, [Mikael Hermansson](https://github.com/mihe) and [Jorrit Rouwe](https://github.com/jrouwe), took this idea one step further: integrating the tool as part of the engine *directly*. There was already a symbiosis between their team and the godot engine, with many features being added to Godot *and* Jolt to accomodate both, but the integration of an official module was no small feat; their pull request ([GH-99895](https://github.com/godotengine/godot/pull/99895)) ended up adding over 500 files and 115 **thousand** lines of code! So while this was one of the most rigorously tested PRs relative to the amount of time it's been up, it'd be impossible for any team to account for *everything* this behemoth introduced, so we eagerly await your feedback (and bug reports) on one of the most highly-requested features of 4.x.

Reviewer: ```suggestion
The Godot Jolt maintainers, [Mikael Hermansson](https://github.com/mihe) and [Jorrit Rouwe](https://github.com/jrouwe), took this idea one step further: integrating the tool as part of the engine *directly*. There was already a symbiosis between their team and the Godot engine, with many features being added to Godot *and* Jolt to accommodate both, but the integration of an official module was no small feat; their pull request ([GH-99895](https://github.com/godotengine/godot/pull/99895)) ended up adding over 500 files and 115 **thousand** lines of code! So while this was one of the most rigorously tested PRs relative to the amount of time it's been up, it'd be impossible for any team to account for *everything* this behemoth introduced, so we eagerly await your feedback (and bug reports) on one of the most highly-requested features of 4.x.
```

> **NOTE:** At time of writing, this does **not** replace Godot Physics as the default 3D physics engine. This is because Jolt Physics have yet to reach full feature-parity with Godot Physics. If your interests/use-case are supported, the tool can be enabled by changing the `physics/3d/physics_engine` project setting to `Jolt Physics`.

Reviewer: ```suggestion
**NOTE:** At time of writing, this does **not** replace Godot Physics as the default 3D physics engine. This is because Jolt Physics has yet to reach full feature-parity with Godot Physics. If your interests/use-case are supported, the tool can be enabled by changing the `physics/3d/physics_engine` project setting to `Jolt Physics`.
```
At least, I *think* it should be "has".

> ### .NET 8.0

Reviewer 2: - Feel free to take anything from https://github.com/godotengine/godot-website/pull/956.

### Improved Scene Tree editor performance

> The amount of performance boosts that've been crammed in between dev 6 and dev 7 has been staggering. Of note are the recent improvements to the engine's core from a multitude of changes to how Strings are processed, spearheaded by [Ivorforce](https://github.com/Ivorforce). But perhaps the most obvious improvement comes from [HP van Braam](https://github.com/godotengine/godot/pull/99700) & his modifications to the scene tree system ([GH-99700](https://github.com/godotengine/godot/pull/99700)). While this will prove beneficial across all projects, these changes result in especially complex scenes having a tenfold performance increase!

Reviewer: ```suggestion
The amount of performance boosts that have been crammed in between dev 6 and dev 7 has been staggering. Of note are the recent improvements to the engine's core from a multitude of changes to how Strings are processed, spearheaded by [Ivorforce](https://github.com/Ivorforce). But perhaps the most obvious improvement comes from [HP van Braam](https://github.com/godotengine/godot/pull/99700) and his modifications to the scene tree system ([GH-99700](https://github.com/godotengine/godot/pull/99700)). While this will prove beneficial across all projects, these changes result in especially complex scenes having a tenfold performance increase!
```

### Documentation tooltips

> ### Baked Shadowmasks for Lightmaps

Reviewer: ```suggestion
### Baked shadowmasks for lightmaps
```

some key features which were already in those snapshots, and are therefore still available for testing in dev 7.

Here are highlights of a few new features in dev 7 that you might find particularly exciting!

### Jolt Physics Module

Ever since its inception in late 2022, [godot-jolt](https://github.com/godot-jolt/godot-jolt) has become the de-facto 3D
physics engine for many of our developers. Much of the history behind why this was the case is documented in [this
proposal](https://github.com/godotengine/godot-proposals/issues/7308) by [Adam Scott](https://github.com/adamscott), but
the main takeaway was a strong push for this tool to be recognized as an official addon. In doing so, users would be
able to find this amazing resource in a way that was promoted by the engine itself; an exciting prospect!

The Godot Jolt maintainers, [Mikael Hermansson](https://github.com/mihe) and [Jorrit Rouwe](https://github.com/jrouwe),
took this idea one step further: integrating the tool as part of the engine *directly*. There was already a symbiosis

> between their team and the godot engine, with many features being added to Godot *and* Jolt to accommodate both, but the

Reviewer: ```suggestion
between their team and the Godot engine, with many features being added to Godot *and* Jolt to accommodate both, but the
```

integration of an official module was no small feat; their pull request
([GH-99895](https://github.com/godotengine/godot/pull/99895)) ended up adding over 500 files and 115 **thousand** lines
of code! So while this was one of the most rigorously tested PRs relative to the amount of time it's been up, it'd be
impossible for any team to account for *everything* this behemoth introduced, so we eagerly await your feedback (and bug
reports) on one of the most highly-requested features of 4.x.

**NOTE:** At time of writing, this does **not** replace Godot Physics as the default 3D physics engine. This is because

> Jolt Physics have yet to reach full feature-parity with Godot Physics. If your interests/use-case are supported, the

Reviewer: ```suggestion
Jolt Physics has yet to reach full feature-parity with Godot Physics. If your interests/use-case are supported, the
```

> **NOTE:** At time of writing, this does **not** replace Godot Physics as the default 3D physics engine. This is because
> Jolt Physics has yet to reach full feature-parity with Godot Physics. If your interests/use-case are supported, the
> tool can be enabled by changing the `physics/3d/physics_engine` project setting to `Jolt Physics`.

Reviewer 3: ```suggestion
**Note:** At time of writing, this does **not** replace Godot Physics as the default 3D physics engine.
The Jolt Physics integration in Godot is considered experimental, and may change in future releases. It also lacks some features of Godot Physics so isn't a full drop-in replacement. If your interests/use-case are supported, the
tool can be enabled by changing the `physics/3d/physics_engine` project setting to `Jolt Physics`.
```

### .NET 8.0

[Paul Joannon](https://github.com/paulloz) and [Raul Santos](https://github.com/raulsntos) have put the final pieces in

> place for supporting .NET 8 in Godot projects ([GH-92131](https://github.com/godotengine/godot/pull/92131) and

Reviewer 3: AFAIK users have been able to use .NET 8 for a long time already, by just specifying it manually in their `.csproj` file.

```suggestion
place for moving both the GodotSharp library and user projects to .NET 8 ([GH-92131](https://github.com/godotengine/godot/pull/92131) and
```

> place for supporting .Net 8 in Godot projects ([GH-92131](https://github.com/godotengine/godot/pull/92131) and

Reviewer: ```suggestion
place for supporting .NET 8 in Godot projects ([GH-92131](https://github.com/godotengine/godot/pull/92131) and
```

> [GH-100195](https://github.com/godotengine/godot/pull/100195)). New projects will automatically use .Net 8 and there is

Reviewer: ```suggestion
[GH-100195](https://github.com/godotengine/godot/pull/100195)). New projects will automatically use .NET 8 and there is
```

> [GH-100195](https://github.com/godotengine/godot/pull/100195)). New projects will automatically use .NET 8 and there is
> an upgrade system in place for existing projects that want to upgrade as well.

Reviewer 4: ```suggestion
[GH-100195](https://github.com/godotengine/godot/pull/100195)). All new projects will use .NET 8 by default and existing projects will automatically update to .Net 8 once opened with this release or any newer 4.4 build.
```

I realized this was wrong once I saw https://github.com/godotengine/godot-website/pull/956

> existing projects will automatically update to .Net 8 once opened with this release or any newer 4.4 build.

Reviewer: ```suggestion
existing projects will automatically update to .NET 8 once opened with this release or any newer 4.4 build.
```

### Improved Scene Tree editor performance

> The amount of performance boosts that've been crammed in between dev 6 and dev 7 has been staggering. Of note are the

Reviewer: ```suggestion
The amount of performance boosts that have been crammed in between dev 6 and dev 7 has been staggering. Of note are the
```

> recent improvements to the engine's core including a many optimizes throughout the String class, spearheaded by

Reviewer 3: ```suggestion
recent improvements to the engine's core including many optimizations throughout the String class, spearheaded by
```

[Ivorforce](https://github.com/Ivorforce). But perhaps the most obvious improvement comes from [HP van
Braam](https://github.com/hpvb) and their modifications to the SceneTree system
([GH-99700](https://github.com/godotengine/godot/pull/99700)). While this will prove beneficial across all projects,
these changes result in especially complex scenes having a tenfold performance in the editor when moving or renaming
nodes in the SceneTree.

### Documentation tooltips

Thanks to [Danil Alexeev](https://github.com/dalexeev) the GDScript code editor will now display a tooltip containing
information about functions, variables, classes, etc. when you hover over them, including the documentation you have
written using our documentation system ([GH-91060](https://github.com/godotengine/godot/pull/91060)). This makes using
the integrated documentation system even more powerful as you no longer have to bounce between the code editor and the
related docs to quickly get information.

> ![Documentation tooltip](/storage/blog/dev-snapshot-godot-4-4-dev-7/documentation-tooltip.png)

Reviewer 3: Should be converted to `.webp` too.

But maybe we can find an example that's a bit more interesting than a placeholder "Method description." on a dummy `method`? I'd show a tooltip for a known API method with documentation.

Reviewer 3: Replaced with this as webp:
![image](https://github.com/user-attachments/assets/ec5d11bf-7cd9-4e9c-85d6-2148c4833cd8)

### Baked Shadowmasks for Lightmaps

Until now, you have always had to choose between fully baked or fully dynamic shadows when using LightmapGI. However,
sometimes you want to have dynamic shadows up close where detail matters, but use static shadows in the distance where a

> low resolution is acceptable. Finally it is possible thanks the the hard work of

Reviewer: ```suggestion
low resolution is acceptable. Finally it is possible thanks to the hard work of
```

- Shaders: Avoid error spam when shaders fail to compile by freeing shader_data version when compilation fails ([GH-100128](https://github.com/godotengine/godot/pull/100128)).
- XR: Allow locking the camera to the `XROrigin3D` for benchmarking or automated testing ([GH-99145](https://github.com/godotengine/godot/pull/99145)).
- XR: OpenXR: Add support for binding modifiers ([GH-97140](https://github.com/godotengine/godot/pull/97140)).

## Changelog

**113 contributors** submitted **263 improvements** for this new snapshot. See our [**interactive changelog**](https://godotengine.github.io/godot-interactive-changelog/#4.4-dev7) for the complete list of changes since the previous 4.4-dev6 snapshot.

This release is built from commit [`46c8f8c5c`](https://github.com/godotengine/godot/commit/46c8f8c5c5874c7c56ea5b1384259de9402d3449).

## Downloads

{% include articles/download_card.html version="4.4" release="dev7" article=page %}

**Standard build** includes support for GDScript and GDExtension.

**.NET build** (marked as `mono`) includes support for C#, as well as GDScript and GDExtension.

> - See also [C# platform support](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/index.html#c-platform-support).

Reviewer 3: That documentation page will need to be updated, CC @raulsntos @paulloz.
For now:
```suggestion
- .NET 8.0 or newer is required for this build, changing the minimal supported version from .NET 6 to 8.
```

Reviewer 2: Yep, here's the PR: https://github.com/godotengine/godot-docs/pull/10427
